### PR TITLE
Fix broken link checker

### PR DESCRIPTION
**Pull request by timoludwig**
### Short description

Fix some problems in the broken link checker

### Proposed changes

- Fix change of pagination size in broken link checker
- Fix pagination in broken link checker
- Show same URLs only once in broken link checker
- Delete outdated link objects
- Translate link checker status
- Fix uncaught TypeError in bulk action script
- Add chunk size option of 500

_Note: I just noticed I forgot to delete old links on the XLIFF import - this I will have to change as well, otherwise outdated links might be shown in the list._

### Resolved issues

Fixes: #1405
Fixes: #1413
Fixes: #1422

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,9 @@ UNRELEASED
 * [ [#1402](https://github.com/digitalfabrik/integreat-cms/issues/1402) ] Also duplicate imprints for new regions
 * [ [#1408](https://github.com/digitalfabrik/integreat-cms/issues/1408) ] Remove duplication of push API tokens for pages during duplication process
 * [ [#1404](https://github.com/digitalfabrik/integreat-cms/issues/1404) ] Fix performance issue for select all on huge page trees
+* [ [#1413](https://github.com/digitalfabrik/integreat-cms/issues/1413) ] Fix change of pagination size in broken link checker
+* [ [#1422](https://github.com/digitalfabrik/integreat-cms/issues/1422) ] Keep page in broken link checker when performing replacement
```
Comment (svenseeberg) [inline CHANGELOG.md:12]: I'm not sure if I understand this line correctly. The broken URL is removed from the list of broken links as soon as it is replaced.

**Review by svenseeberg — COMMENTED** [review]
Thanks, I cannot see any immediate problems with your code changes. However, I have some more general comments and I stumbled upon one stack trace while updating a broken link with 3 occurrences but I'm not sure how to reproduce:

```
May 14 12:19:39 ERROR django.request - Internal Server Error: /augsburg/analytics/linkcheck/invalid/3/
Traceback (most recent call last):
  File "/home/sven/Code/integreat-cms-django/.venv/lib64/python3.9/site-packages/django/db/backends/base/base.py", line 242, in _commit
    return self.connection.commit()
psycopg2.errors.ForeignKeyViolation: update or delete on table "linkcheck_url" violates foreign key constraint "linkcheck_link_url_id_c5f791e9_fk" on table "linkcheck_link"
DETAIL:  Key (id)=(1) is still referenced from table "linkcheck_link".


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/sven/Code/integreat-cms-django/.venv/lib64/python3.9/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/home/sven/Code/integreat-cms-django/.venv/lib64/python3.9/site-packages/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/sven/Code/integreat-cms-django/.venv/lib64/python3.9/site-packages/django/views/generic/base.py", line 70, in view
    return self.dispatch(request, *args, **kwargs)
  File "/home/sven/Code/integreat-cms-django/.venv/lib64/python3.9/site-packages/django/views/generic/base.py", line 98, in dispatch
    return handler(request, *args, **kwargs)
  File "/home/sven/Code/integreat-cms-django/integreat_cms/cms/views/linkcheck/linkcheck.py", line 161, in post
    translation.links.all().delete()
  File "/home/sven/Code/integreat-cms-django/.venv/lib64/python3.9/site-packages/django/db/models/query.py", line 746, in delete
    deleted, _rows_count = collector.delete()
  File "/home/sven/Code/integreat-cms-django/.venv/lib64/python3.9/site-packages/django/db/models/deletion.py", line 435, in delete
    signals.post_delete.send(
  File "/home/sven/Code/integreat-cms-django/.venv/lib64/python3.9/site-packages/django/db/transaction.py", line 246, in __exit__
    connection.commit()
  File "/home/sven/Code/integreat-cms-django/.venv/lib64/python3.9/site-packages/django/utils/asyncio.py", line 33, in inner
    return func(*args, **kwargs)
  File "/home/sven/Code/integreat-cms-django/.venv/lib64/python3.9/site-packages/django/db/backends/base/base.py", line 266, in commit
    self._commit()
  File "/home/sven/Code/integreat-cms-django/.venv/lib64/python3.9/site-packages/django/db/backends/base/base.py", line 242, in _commit
    return self.connection.commit()
  File "/home/sven/Code/integreat-cms-django/.venv/lib64/python3.9/site-packages/django/db/utils.py", line 90, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/home/sven/Code/integreat-cms-django/.venv/lib64/python3.9/site-packages/django/db/backends/base/base.py", line 242, in _commit
    return self.connection.commit()
django.db.utils.IntegrityError: update or delete on table "linkcheck_url" violates foreign key constraint "linkcheck_link_url_id_c5f791e9_fk" on table "linkcheck_link"
DETAIL:  Key (id)=(1) is still referenced from table "linkcheck_link".
```

File: CHANGELOG.md (hunk 1 above)
Comment (timoludwig) [reply CHANGELOG.md:12]: Yeah, it's ambiguous, in this case I meant the pagination page... I'll adapt the wording.

File: integreat_cms/cms/views/linkcheck/linkcheck.py [hunk 2]
```diff
@@ -82,28 +142,23 @@ def post(self, request, *args, **kwargs):
...
                 # Get all current translations with the same url
-                translations = [
-                    item["translation"]
-                    for item in links
-                    if item["link"].url == old_link.url
-                ]
+                translations = [link.content_object for link in old_url.links.all()]
                 # Replace the old urls with the new urls in the content
                 for translation in translations:
                     translation.content = re.sub(
```
[the remark this answers is not in the document]
Comment (timoludwig) [reply integreat_cms/cms/views/linkcheck/linkcheck.py]: Good point, I replaced the regex handling by the function with [lxml.html.rewrite_links()](https://lxml.de/lxmlhtml.html).

File: integreat_cms/locale/de/LC_MESSAGES/django.po [hunk 3]
```diff
@@ -1183,6 +1183,126 @@ msgstr "Monatlich"
 msgid "Yearly"
 msgstr "Jährlich"
 
+#: cms/constants/linkcheck.py:4
```
[the remark this answers is not in the document]
Comment (timoludwig) [reply integreat_cms/locale/de/LC_MESSAGES/django.po:1186]: Sounds reasonable, I removed the translation of status codes.

**Comment by timoludwig** [comment]
> However, I have some more general comments and I stumbled upon one stack trace while playing updating a broken link with 3 occurrences but I'm not sure how to reproduce

I think this might have been caused by a race condition - the post_save signal is called asynchronously in a separate thread - when old links are deleted while the post_save signal is running, the database integrity seems to be violated.
I think I solved this by acquiring [linkcheck's internal lock](https://github.com/DjangoAdminHackers/django-linkcheck/blob/dfa053fbebd832e17de477dd9653472b775d0261/linkcheck/__init__.py#L7) for this - could you play around with it again and see whether you observe any errors?

**Review by svenseeberg — APPROVED** [review]

**Comment by svenseeberg** [comment]
I tested a little more and could not find issues.

**Pull request merged**